server/shared/types/terms/term.ts

```diff
@@ -59,29 +57,28 @@ export type GroupSetEntry = BaseGroupSet & {
 	is_subcondition?: boolean
 }
 
-export type CustomGroupSetting = {
+export type CustomQGroupSetting = {
 	/** When “predefined_groupset_idx” is undefined, will use this set of groups.
 	This is a custom set of groups either copied from predefined set, or created with UI.
 	Custom set definition is the same as a predefined set. */
 	customset: BaseGroupSet
-	disabled?: boolean
+	/** if false, not applied */
 	inuse?: boolean
-	lst?: GroupSetEntry[] // quick-fix
 }
 
-export type PredefinedGroupSetting = {
-	/** If true, apply and will require the following attributes */
-	inuse?: boolean
-	disabled?: boolean
-	useIndex?: number
-	/**Value is array index of term.groupsetting.lst[] */
+export type PredefinedQGroupSetting = {
+	/** If .inuse true, apply and will require predefined_groupset_idx */
+	/** Value is array index of term.groupsetting.lst[] */
 	predefined_groupset_idx: number
-	lst: GroupSetEntry[]
+	inuse: true
 }
 
-export type EmptyGroupSetting = {
-	inuse?: false
-	disabled?: true
+export type QGroupSetting = PredefinedQGroupSetting | CustomQGroupSetting
+
+export type TermGroupSetting = {
+	/** if there are only two values, means groupsetting definition is not applicable for the term */
+	disabled?: boolean
```

Comment:
I think `disabled` should be required

Reply:
We chatted about about this property this morning. I don't think always present but I'll double check.